### PR TITLE
Add docker

### PR DESCRIPTION
**Pull request by johnrazeur**
Add docker to the boilerplate.

Readme has been updated.

### CODE REVIEW

**Comment by mattdamon108** [comment]
Hi,

Thank you for your PR.
But, Can you create a PR to another branch not `master`?

**Comment by johnrazeur** [comment]
Done, 

Sorry for the inconvenience :)

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+FROM golang:1.12 as builder
+
+WORKDIR /go/src/github.com/mattdamon108/go-graphql-api-boilerplate
```
Comment (mattdamon108) [inline Dockerfile]: I prefer to use `go modules`. How about changing to enable to use `go modules`?

```suggestion
WORKDIR /usr/app
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+COPY . .
+RUN go get -d -v ./...
```
Comment (mattdamon108) [inline Dockerfile]: If we use `go modules`, it doesn't need any more.
```suggestion
```

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o go-graphql-api-boilerplate .
+RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o migrate ./migrations/
+
+# Multi stage
+
+FROM alpine:latest  
+RUN apk --no-cache add ca-certificates
+ENV PATH /root
+WORKDIR /root/
+COPY --from=builder /go/src/github.com/mattdamon108/go-graphql-api-boilerplate/go-graphql-api-boilerplate .
```
Comment (mattdamon108) [inline Dockerfile]:
```suggestion
COPY --from=builder /usr/app/go-graphql-api-boilerplate .
```

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+COPY --from=builder /go/src/github.com/mattdamon108/go-graphql-api-boilerplate/migrate .
```
Comment (mattdamon108) [inline Dockerfile]:
```suggestion
COPY --from=builder /usr/app/migrate .
```

File: README.md [hunk 5]
```diff
@@ -56,13 +56,23 @@ func ConnectDB() (*DB, error) {
...
 
 ### Run the server
 
 ```shell
 $ go run server.go
 ```
+or with docker
+```
+$ docker build -t go-graphql-api .
+$ docker run -p 8080:8080 go-graphql-api
```
Comment (mattdamon108) [inline README.md]:
```suggestion
$ docker run --rm -d -p 8080:8080 go-graphql-api
```

**Pull request merged**